sqlglot/dialects/clickhouse.py

```diff
@@ -1045,6 +1045,20 @@ def combinedparameterizedagg_sql(self, expression: exp.CombinedParameterizedAgg)
         def placeholder_sql(self, expression: exp.Placeholder) -> str:
             return f"{{{expression.name}: {self.sql(expression, 'kind')}}}"
 
+        def subquery_sql(
+            self, expression: exp.Subquery, sep: str = " AS ", wrap_left="(", wrap_right=")"
+        ) -> str:
+            alias = self.sql(expression, "alias")
+            if not alias and isinstance(expression.this, exp.Subquery):
+                return super().subquery_sql(expression, wrap_left="", wrap_right="")
+            if (
+                not alias
+                and isinstance(expression.this, exp.Select)
+                and expression.parent_select is None
+            ):
+                return super().subquery_sql(expression, wrap_left="", wrap_right="")
```

Comment:
Can you elaborate on what's happening here? What is this trying to achieve exactly? On a side note, it doesn't seem like there's a good reason to have two branches, we can simplify by merging them.